tools/collect-manifests.sh

```diff
@@ -54,51 +59,113 @@ elif [[ -f $TARFILE ]]; then
+             bin/kustomize build config/begin | yq eval 'select(.kind != "Namespace" and .kind != "CustomResourceDefinition")' > "$SUBMOD_DIR/$SUBMODULE.yaml"
+             bin/kustomize build config/begin | yq eval 'select(.kind == "CustomResourceDefinition")' > "$SUBMOD_DIR/$SUBMODULE-crds.yaml"
+         fi
+         if [[ -d config/begin-examples ]]; then
+             bin/kustomize build config/begin-examples > "$SUBMOD_DIR/$SUBMODULE-examples.yaml"
+         fi
+         if [[ -d config/prometheus ]]; then
+             bin/kustomize build config/prometheus > "$SUBMOD_DIR/$SUBMODULE-prometheus.yaml"
+         fi
+         if [[ -d deploy/kubernetes/begin ]]; then
+             # Remove the namespace from the manifest, because this manifest is
+             # used not only to deploy but also to undeploy.
+             # The namespace will be created by the ArgoCD Application resource,
+             # and nothing will delete it.
+             bin/kustomize build deploy/kubernetes/begin | yq eval 'select(.kind != "Namespace")' > "$SUBMOD_DIR/$SUBMODULE.yaml"
+         fi
+        )
+    done
+}
+
+walk_overlays() {
+    if [[ ! -f $SYSTEMS_YAML ]]; then
+        echo "Unable to find $SYSTEMS_YAML"
+        exit 1
     fi
-done
 
-for SUBMODULE in $DO_MODULES; do
-    echo "Collecting the manifest from $SUBMODULE"
-    mkdir "$TREEDIR/$SUBMODULE"
-    (cd "$SUBMODULE" || exit 1
-     if [[ -d config/begin ]]; then
-         # Remove the namespace from the manifest, because this manifest is
-         # used not only to deploy but also to undeploy.
-         # The namespace will be created by the ArgoCD Application resource,
-         # and nothing will delete it.
-         # Place the CRDs in a separate manifest.
-         bin/kustomize build config/begin | yq eval 'select(.kind != "Namespace" and .kind != "CustomResourceDefinition")' > "$TREEDIR/$SUBMODULE/$SUBMODULE.yaml"
-         bin/kustomize build config/begin | yq eval 'select(.kind == "CustomResourceDefinition")' > "$TREEDIR/$SUBMODULE/$SUBMODULE-crds.yaml"
-     fi
-     if [[ -d config/begin-examples ]]; then
-         bin/kustomize build config/begin-examples > "$TREEDIR/$SUBMODULE/$SUBMODULE-examples.yaml"
-     fi
-     if [[ -d config/prometheus ]]; then
-         bin/kustomize build config/prometheus > "$TREEDIR/$SUBMODULE/$SUBMODULE-prometheus.yaml"
-     fi
-     if [[ -d deploy/kubernetes/begin ]]; then
-         # Remove the namespace from the manifest, because this manifest is
-         # used not only to deploy but also to undeploy.
-         # The namespace will be created by the ArgoCD Application resource,
-         # and nothing will delete it.
-         bin/kustomize build deploy/kubernetes/begin | yq eval 'select(.kind != "Namespace")' > "$TREEDIR/$SUBMODULE/$SUBMODULE.yaml"
-     fi
-    )
-done
+    if [[ $SYSTEM_TYPE == 'kind' ]]; then
+        point_to_overlay kind
+        collect_manifest
+    else
+        point_to_overlay rabbit-tds
```

Comment:
Couldn't this also be rabbit-htx? If so, this should be parameterized, no?

Reply:
I picked rabbit-tds because it doesn't include cluster-specific overlays, like rabbit-htx does.   It shouldn't be parameterized. Instead, the per-cluster customizing should happen in the gitops repo.